test/abstractarray.jl

```diff
@@ -840,3 +840,9 @@ end
 @testset "checkbounds_indices method ambiguities #20989" begin
     @test Base.checkbounds_indices(Bool, (1:1,), ([CartesianIndex(1)],))
 end
+
+@testset "ImageCore #40" begin
+    Base.convert(::Type{Array{T,n}}, a::Array) where {T<:Number,n} =
```

Comment:
why is defining a method like this necessary to reproduce the bug? also, where is `dest` supposed to be coming from?